crates/witgen/src/cert_validity_provider.rs

```diff
@@ -0,0 +1,32 @@
+use alloy_primitives::{BlockNumber, B256};
+use canoe_provider::{CanoeInput, CanoeProvider};
+use hokulea_proof::eigenda_blob_witness::EigenDABlobWitnessData;
+
+/// Populate canoe proof into cert validity. It is placed inside the normal crate because canoe
+/// currently assumes serde_json for serialization.
+pub async fn populate_cert_validity_to_witness(
+    witness: &mut EigenDABlobWitnessData,
```

Comment:
what are the assumptions on witness? Is it already half populated? Or is it just initialized and all fields are initialized by this function? If so, why not just return it as a value instead of passing a reference?

Typically when passing a &mut ref its best to describe why, and what are the assumptions around its use.